src/routes/v1/capprevision_test.go

```diff
@@ -2,36 +2,202 @@ package v1_test
+	labelKey              = "key"
+	labelValue            = "value"
 )
 
+func setupCappRevisions() {
+	createTestNamespace(cappRevisionNamespace)
+	createTestCappRevision(cappRevisionName+"-1", cappRevisionNamespace, map[string]string{labelKey + "-1": labelValue + "-1"}, map[string]string{})
+	createTestCappRevision(cappRevisionName+"-2", cappRevisionNamespace, map[string]string{labelKey + "-2": labelValue + "-2"}, map[string]string{})
+}
+
 func TestGetCappRevisions(t *testing.T) {
-	request, _ := http.NewRequest("GET", "/v1/namespaces/test-namespace/capprevisions/", nil)
-	writer := httptest.NewRecorder()
-	router.ServeHTTP(writer, request)
-
-	assert.Equal(t, http.StatusOK, writer.Code)
-	var response types.CappRevisionList
-	err := json.Unmarshal(writer.Body.Bytes(), &response)
-	assert.NoError(t, err)
-	assert.GreaterOrEqual(t, len(response.CappRevisions), 1)
-	assert.GreaterOrEqual(t, response.Count, 1)
+	type selector struct {
+		keys   []string
+		values []string
+	}
+
+	type requestParams struct {
+		namespace     string
+		labelSelector selector
+	}
+
+	type want struct {
+		statusCode int
+		response   types.CappRevisionList
+	}
+
+	cases := map[string]struct {
+		requestParams requestParams
+		want          want
+	}{
+		"ShouldSucceedGettingCappRevisions": {
+			requestParams: requestParams{
+				namespace: cappRevisionNamespace,
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: types.CappRevisionList{Count: 2, CappRevisions: []cappv1alpha1.CappRevision{
+					resources.GetBareCappRevision(cappRevisionName+"-1", cappRevisionNamespace, map[string]string{labelKey + "-1": labelValue + "-1"}, map[string]string{}),
+					resources.GetBareCappRevision(cappRevisionName+"-2", cappRevisionNamespace, map[string]string{labelKey + "-2": labelValue + "-2"}, map[string]string{}),
+				}},
+			},
+		},
+		"ShouldFailWithBadRequestInvalidURI": {
+			requestParams: requestParams{
+				namespace: "",
+			},
+			want: want{
+				statusCode: http.StatusBadRequest,
+				response:   types.CappRevisionList{},
+			},
+		},
+		"ShouldSucceedGettingCappRevisionsWithLabelSelector": {
+			requestParams: requestParams{
+				namespace: cappRevisionNamespace,
+				labelSelector: selector{
+					keys:   []string{labelKey + "-1"},
+					values: []string{labelValue + "-1"},
+				},
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: types.CappRevisionList{Count: 1, CappRevisions: []cappv1alpha1.CappRevision{
+					resources.GetBareCappRevision(cappRevisionName+"-1", cappRevisionNamespace, map[string]string{labelKey + "-1": labelValue + "-1"}, map[string]string{}),
+				}},
+			},
+		},
+		"ShouldFailGettingCappRevisionsWithInvalidLabelSelector": {
+			requestParams: requestParams{
+				namespace: cappRevisionNamespace,
+				labelSelector: selector{
+					keys:   []string{labelKey + "-1"},
+					values: []string{labelValue + " 1"},
+				},
+			},
+			want: want{
+				statusCode: http.StatusBadRequest,
```

Comment:
it cant be bad request without throw an error

Reply:
this is a bug at our side if we have bad requeset we must return an error

Reply:
It does return an error from [here](https://github.com/dana-team/platform-backend/blob/c29e1994804687956fa629240b03fe72487cc719/src/controllers/capprevision.go#L51-L56), I just don't assert on it because I only assert the response on success status codes.